### PR TITLE
updated digital interface to support larger amount of item/fluid/energy

### PR DESCRIPTION
**Pull request by galyfray**
One of my friend pointed out that when using the mulktiblock battery the number showed by the digital interface may  overflow from where it should be:

i fixed this by changing a little the computation in order to make easier to add new  unit tier (yeah not sure about how to says this) and adding the Giga Terra and Peta prefix hopping it should be enough

### CODE REVIEW

**Comment by galyfray** [comment]
ohhhh no I didn't pay attention about it but idea has reformated everything what a hell hope it doesn't annoy you too much I'm so sorry about that

**Review by DStrand1 — APPROVED** [review]
Seems like a good fix. With regards to the reformatting, it looks like this file really needed it, so no problem.

**Review by Yefancy — APPROVED** [review]

File: src/main/java/gregicadditions/covers/CoverDigitalInterface.java [hunk 1]
```diff
@@ -962,30 +967,40 @@ private void renderItemMode(int slot) {
...
         }
-        return new DecimalFormat("#.#").format(number * 1.0f / 1000000) + units[2][unit];
+        int i = 1;
+
+        while (number / 10000000 != 0 && i < units.length) {
+            number = number / 1000;
+            ++i;
+        }
+
+        return new DecimalFormat("#.#").format(number * 1.0f / 1000) + units[i][unit];
```
Comment (Yefancy) [inline src/main/java/gregicadditions/covers/CoverDigitalInterface.java:1004]: I think 10,000,000 is wrong here?
E.g. After calculation, 1,000,000 (eu) => 1,000.0KEU, but the expected result should be 1.0MEU.

Comment (Yefancy) [reply src/main/java/gregicadditions/covers/CoverDigitalInterface.java:1004]: If you purpose to keep a little more length of the text, for the sake of looking good, I think it's OK

Comment (galyfray) [reply src/main/java/gregicadditions/covers/CoverDigitalInterface.java:1004]: yeah I think that 1000KEU can be a bit nicer than directly going to MEU even if that wasn't planned if nobody disagree with this then let's go with this else I will remove this 0.

**Comment by Yefancy** [comment]
I think it could be merged now.  And i have a pr to do recently that I could handle the conflicts with this file in advance

**Review by hjae78 — APPROVED** [review]

**Pull request merged**